cpp/tests/io/parquet_test.cpp

```diff
@@ -846,18 +891,22 @@ TEST_F(ParquetChunkedWriterTest, ReadRowGroupsError)
   auto table1 = create_random_fixed_table<int>(5, 5, true);
 
   auto filepath = temp_env->get_temp_filepath("ChunkedRowGroupsError.parquet");
-  cudf_io::write_parquet_chunked_args args{cudf_io::sink_info{filepath}};
+  cudf_io::chunked_parquet_writer_options args =
+    cudf_io::chunked_parquet_writer_options::builder(cudf_io::sink_info{filepath});
   auto state = cudf_io::write_parquet_chunked_begin(args);
   cudf_io::write_parquet_chunked(*table1, state);
   cudf_io::write_parquet_chunked_end(state);
 
-  cudf_io::read_parquet_args read_args{cudf_io::source_info{filepath}};
-  read_args.row_groups = {{0, 1}};
-  EXPECT_THROW(cudf_io::read_parquet(read_args), cudf::logic_error);
-  read_args.row_groups = {{-1}};
-  EXPECT_THROW(cudf_io::read_parquet(read_args), cudf::logic_error);
-  read_args.row_groups = {{0}, {0}};
-  EXPECT_THROW(cudf_io::read_parquet(read_args), cudf::logic_error);
+  cudf_io::parquet_reader_options read_opts =
+    cudf_io::parquet_reader_options::builder(cudf_io::source_info{filepath})
+      .set_row_groups({{0, 1}});
+  EXPECT_THROW(cudf_io::read_parquet(read_opts), cudf::logic_error);
+  read_opts =
+    cudf_io::parquet_reader_options::builder(cudf_io::source_info{filepath}).set_row_groups({{-1}});
```

Comment:
We could just set the row groups directly in the existing opts object. Seems like a cleaner/faster option to me.